ingenialink/ethernet/servo.py

```diff
@@ -143,13 +143,33 @@ def _send_mcb_frame(self, cmd, reg, subnode, data=None):
             except socket.error as e:
                 raise ILIOError("Error sending data.") from e
             try:
-                response = self.socket.recv(ETH_BUF_SIZE)
-            except socket.timeout as e:
-                raise ILTimeoutError("Timeout while receiving data.") from e
-            except socket.error as e:
-                raise ILIOError("Error receiving data.") from e
-        except (ILIOError, ILTimeoutError) as e:
-            raise e
+                return self.__receive_mcb_frame(reg, self.socket)
+            except ILWrongRegisterError as e:
+                logger.error(e)
+                return self.__receive_mcb_frame(reg, self.socket)
         finally:
             self._lock.release()
+
+    @staticmethod
+    def __receive_mcb_frame(reg: int, conn_socket: socket.socket) -> bytes:
```

Comment:
Why not use the instance attribute socket?

Reply:
I don't have a reason really, I moved the code to a function and I sent the variables as arguments